alibi/explainers/cfrl_base.py

```diff
@@ -839,7 +832,7 @@ def explain(self,
 
         # Perform prediction in mini-batches.
         n_minibatch = int(np.ceil(X.shape[0] / batch_size))
-        all_results: Dict[str, np.ndarray] = {}
+        all_results: Dict[str, Optional[np.ndarray]] = {}
```

Comment:
Changed this to `Optional` because `C` can be `None`. However, this complicates typing further down as all other entries cannot be `None`, but the type checker doesn't know that. A `TypedDict` might be a better fit here in the future.